lib_eio/task_pool.ml

```diff
@@ -0,0 +1,44 @@
+type 'a task = unit -> 'a
+
+type runner = unit task -> unit
+
+type t = {
+  runners : runner Pool.t;
+}
+
+let create ~max_domains domain_mgr : t =
+  let alloc () : runner =
+    let s : unit task Stream.t = Stream.create 0 in
+    Switch.run (fun sw ->
+        Fiber.fork ~sw (fun () ->
```

Comment:
This pattern doesn't do anything useful, because the switch will wait for the fiber to finish.